src/ui.rs

```diff
@@ -338,23 +340,36 @@ impl<UI: UserInterface> game::Loop<UI> for Loop<UI> {
             interface.on_event(event, cursor_position, messages)
         });
 
-        let new_cursor = interface.draw(
+        let new_ui_cursor = interface.draw(
             &mut self.renderer,
             &mut window.frame(),
             cursor_position,
         );
 
         self.cache = Some(interface.cache());
 
-        if new_cursor != self.mouse_cursor {
-            if new_cursor == MouseCursor::OutOfBounds {
+        if new_ui_cursor != self.ui_cursor {
+            if new_ui_cursor == MouseCursor::OutOfBounds {
                 input.update(input::Event::Mouse(mouse::Event::CursorReturned));
-            } else if self.mouse_cursor == MouseCursor::OutOfBounds {
+            } else if self.ui_cursor == MouseCursor::OutOfBounds {
                 input.update(input::Event::Mouse(mouse::Event::CursorTaken));
             }
 
-            window.update_cursor(new_cursor.into());
-            self.mouse_cursor = new_cursor;
+            self.ui_cursor = new_ui_cursor;
+        }
+        let new_game_cursor = ui.cursor_icon();
+        if new_game_cursor != self.game_cursor {
+            self.game_cursor = new_game_cursor;
+        }
```

Comment:
We may be able to move this inside the `if` branch below.

Reply:
Fixed in 13429f2758978231a7888db702c940c4e6a8e913.